### DISCUSSION TITLE
How can I add metadata based on dbt model/source in an @dbt_assets function?

### DISCUSSION THREAD

**Discussion by seandavi** [dagster-io/dagster#18825] [Q&A] [answered]
With a function like:

```
@dbt_assets(manifest=dbt_manifest_path)
def all_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
    yield from dbt.cli(["build"], context=context).stream()
```

Is it possible to grab the table/view name and inject a call to a bigquery API or a duckdb database to gather additional metadata to each dbt asset, like row count, schema, etc?

**Comment by ZacharyMcGuire** [comment, marked as answer]
Still learning both Dagster & dbt myself, so there might be a better way, but hope this helps. 😃 

I believe this is where you'll want to use a custom `DagsterDbtTranslator` class and access the `dbt_resource_props` property. Take a look at the docs below, which should hopefully put you on the right path.
https://docs.dagster.io/integrations/dbt/reference#customizing-metadata

There's an example of one here which constructs the asset key as `database / schema / object_name` which I quite like. Sounds like they're, at least some of, the props you're looking for.
https://github.com/dagster-io/dagster-open-platform/blob/main/dagster_open_platform/utils/dbt_helpers.py#L20
